### PR TITLE
Replaced Raphael with Snap.svg to allow for allow for extra features …

### PR DESCRIPTION
**Pull request by StefRave**
…like styling

Removed the dependency on the Daniel font. The font can be set using style sheets. However
this introduced a requirement to delay rendering until the font is loaded completely so that
text size can be calculated accurately.

### CODE REVIEW

**Comment by bramp** [comment]
Very nice! I quickly scanned the change, but going to wait until the weekend to fully review/pull.

I fully want to do this migration, but some quick comments/concerns (off the top of my head)
1) I would like to make the loading of the hand drawn font seamless. So no work for the user other than having the font in the right path.
2) Is it possible to put your snap.svg work into a separate theme, so Rapheal and snap can coexist? I don't know what my migration plan is, but is it possible to offer both for a while?
3) I will do a major version bump after I pull this change.

**Comment by StefRave** [comment]
No hurry, I don't have any time to do anything the next week
1) I understand, and agree. I could not find a quick solution a short amount of time, but it should be possible.
2) It is possible, but I am not sure if it is worth the hassle. Having to maintain Raphael & Snap.SVG in one codebase can be a challenge. Putting 1.x on a branch will keep code more clean and allow more innovation.
3) I think a is a good solution. Maintain the Raphael version for a while and focus on new features for the SVG version. I think we can make upgrading to the new version easy, but I don't know how this will affect old browser support. Probably >=IE9 will be possible.

**Comment by schrepfler** [comment]
Nice effort @StefRave!

File: src/sequence-diagram.js [hunk 1]
```diff
@@ -79,13 +78,17 @@
 /******************
  * Raphaël extras
  ******************/
+	Snap.plugin(function (Snap, Element, Paper, global, Fragment) {
+		Snap.newmethod = function () { };
+		Element.prototype.newmethod = function () { };
+		Paper.prototype.newmethod = function () { };
```
Comment (bramp) [inline src/sequence-diagram.js]: The above three lines aren't needed. (they are copied and pasted from a demo)

File: test/test.html [hunk 2]
```diff
@@ -17,15 +50,16 @@
     	</textarea>
     	<button id="parse" type="button">Click Me!</button>  
 
-    	<div id="diagram"></div>
+    	<svg id="diagram"></svg>
```
Comment (bramp) [inline test/test.html]: Changing this to a svg, breaks the API a little bit.

**Comment by bramp** [comment]
I've pulled your code into a new branch, and started hacking on it. There are some inconsitances with the old vs new rendering. I'm also refactoring this so there is a RaphaelTheme and a SnapTheme. It was quite easy to keep both, without duplicating code. This is mainly useful so I can easily compare old to new (to check for regressions). I'll update you shortly.

**Comment by bramp** [comment]
I've pulled this into a new SVG branch: https://github.com/bramp/js-sequence-diagrams/tree/svg and started to heavily refactor to allow both Raphael and Snapsvg at the same (if you so wish). I had to cleanup some of this new snapsvg code to match the existing API correctly (for example, snapsvg assumed the container was a SVG, where previous Raphael accepted div/td/etc). Still a work in progress, but getting there. I will close this PR, and track the work in #127

**Pull request closed without merging**